### DISCUSSION TITLE
filter low abundant ASVs

### DISCUSSION THREAD

**Discussion by jamorillo** [ChiLiubio/microeco#430] [Ideas]
Hello,
I have a question about the `filter_taxa` function. The idea is to remove low-frequency ASVs to mitigate the effect of index jumping in Illumina metabarcoding dfata. For example, `microeco_object$filter_taxa(rel_abund = 0.0001)` would filter out ASVs that are <0.01% in the entire dataset. But wouldn’t it be more accurate to remove those that are <0.01% in each sample? If so, is this possible with microeco? I ask because, in large datasets with different sample types, filtering across all samples could eliminate ASVs that represent real and important diversity in certain samples.

Does this make sense to you? thanks! Not really sure about the best approach, I have seen both in articles

**Comment by ChiLiubio** [comment]
Hi. Thanks! To temporarily do this, we can use a way like this:
```
library(microeco)
library(magrittr)

data(sample_info_16S)
data(otu_table_16S)
data(taxonomy_table_16S)
dataset <- microtable$new(sample_table = sample_info_16S, otu_table = otu_table_16S, tax_table = taxonomy_table_16S)
dataset$tidy_dataset()
dataset

# use cal_abund to get relative abundance of each feature
d1 <- clone(dataset)
d1$add_rownames2taxonomy("OTU")
d1$cal_abund(select_cols = "OTU", rel = TRUE)

# judge whether there is low abundance for a OTU
filter_logic <- apply(d1$taxa_abund$OTU, 1, function(x){any(x < 0.0001)})
dataset$otu_table %<>% .[!filter_logic, ]
dataset$tidy_dataset()
dataset
```
The results show that most of OTUs (from 13628 to 10 remained) have a low abundance (< 0.0001) in at least one sample.

Chi

**Reply by jamorillo** [reply]
Aha. Great, Will have a look in my data, thanks!!
